sound/soc/sof/ipc4-topology.c

```diff
@@ -736,10 +743,63 @@ static int sof_ipc4_widget_setup_comp_mixer(struct snd_sof_widget *swidget)
 err:
 	sof_ipc4_free_audio_fmt(&mixer->available_fmt);
 	kfree(mixer);
+	return ret;
+}
+
+static int sof_ipc4_widget_setup_comp_src(struct snd_sof_widget *swidget)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct sof_ipc4_src *src;
+	int ret;
+
+	dev_dbg(scomp->dev, "Updating IPC structure for %s\n", swidget->widget->name);
+
+	src = kzalloc(sizeof(*src), GFP_KERNEL);
+	if (!src)
+		return -ENOMEM;
+
+	swidget->private = src;
+
+	/*
+	 * only input format and sink rate are required to be sent to the FW for SRC,
+	 * so no need to get output format like copier.
+	 */
+	ret = sof_ipc4_get_audio_fmt(scomp, swidget, &src->available_fmt, false);
```

Comment:
Or rather use the same comment as other modules where the output format is ignored?
pga, mixer:
`/* The out_audio_fmt in topology is ignored as it is not required to be sent to the FW */`
The note about sink rate does not help much as it is not visible in the code, it is parsed from the topology

Reply:
sure, thanks.